cmds/core/more/more.go

```diff
@@ -23,43 +23,50 @@ package main
 import (
 	"bufio"
 	"fmt"
+	"io"
 	"log"
 	"os"
 
 	flag "github.com/spf13/pflag"
 )
 
 var lines = flag.Int("lines", 40, "screen size in number of lines")
+var errOnlyOneFile = fmt.Errorf("more can only take one file")
+var errLinesMustBePositive = fmt.Errorf("lines must be positive")
 
-func main() {
-	flag.Parse()
-	if flag.NArg() != 1 {
-		log.Fatal("more can only take one file")
+func run(stdin io.Reader, stdout io.Writer, lines int, args []string) error {
+	if len(args) != 1 {
+		return errOnlyOneFile
 	}
-	if *lines <= 0 {
-		log.Fatal("lines must be positive")
+	if lines <= 0 {
+		return errLinesMustBePositive
```

Comment:
could this be
return fmt.Errorf("%d: %w", lines, errLinesMustBePositive)

sometimes it's nice to see what it thought you passed it, since a shell script may have a bug and this helps find it.

Reply:
Sure, updated.